delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/S3FileOperations.scala

```diff
@@ -78,28 +77,30 @@ object S3FileOperations {
 
     override def register(bucket: String, path: Uri.Path): IO[S3FileMetadata] = {
       for {
-        _           <- log.info(s"Fetching attributes for S3 file. Bucket $bucket at path $path")
-        resp        <- client.headObject(bucket, path.toString())
-        contentType <- parseContentType(resp.contentType())
-        metadata    <- mkS3Metadata(bucket, path, resp, contentType)
+        _        <- log.info(s"Fetching attributes for S3 file. Bucket $bucket at path $path")
+        resp     <- client.headObject(bucket, path.toString())
+        metadata <- mkS3Metadata(bucket, path, resp)
       } yield metadata
     }
       .onError { e =>
         log.error(e)(s"Failed fetching required attributes for S3 file registration. Bucket $bucket and path $path")
       }
 
-    private def parseContentType(raw: String): IO[ContentType] =
-      ContentType.parse(raw).map(_.pure[IO]).getOrElse(IO.raiseError(InvalidContentType(raw)))
+    private def parseContentType(raw: Option[String]): IO[ContentType] = {
+      IO.fromOption(raw)(MissingContentType)
+        .flatMap(raw => ContentType.parse(raw).map(_.pure[IO]).getOrElse(IO.raiseError(InvalidContentType(raw))))
+    }
```

Comment:
Whilst this is just what was here before, I think this doesn't match what is supposed to be happening